internal/pipe/cask/templates/cask.rb

```diff
@@ -0,0 +1,209 @@
+# This file was generated by GoReleaser. DO NOT EDIT.
+cask "{{ .Name }}" do
+  desc "{{ .Description }}"
+  homepage "{{ .Homepage }}"
+  version "{{ .Version }}"
+  {{- with .License }}
+  license "{{ .}}"
+  {{- end }}
+
+  livecheck do
+    skip "Auto-generated on release."
+  end
+
+  {{ with .Binary }}
+  binary "{{ .}}"
+  {{- end }}
+  {{- with .Manpage }}
+  manpage "{{ .}}"
+  {{- end }}
+  {{- with .Completions.Bash }}
+  bash_completions "{{ . }}"
+  {{- end }}
+  {{- with .Completions.Zsh }}
+  zsh_completions "{{ . }}"
+  {{- end }}
+  {{- with .Completions.Fish }}
+  fish_completions "{{ . }}"
+  {{- end }}
+
+  {{- with .Dependencies }}
+  {{ range  . }}
+  {{- with .Cask }}
+  depends_on cask: "{{ . }}"
+  {{- end }}
+  {{- with .Formula }}
+  depends_on formula: "{{ . }}"
+  {{- end }}
+  {{- end }}
+  {{- end }}
+
+  {{ if and .MacOSPackages .LinuxPackages }}
+  on_macos do
+  {{- include "macos_packages" . | indent 2 }}
+  end
+
+  on_linux do
+  {{- include "linux_packages" . | indent 2 }}
+  end
+  {{ end }}
+
+  {{ if and (.MacOSPackages) (not .LinuxPackages) }}
+  {{- template "macos_packages" . }}
+  {{ end }}
+
+  {{ if and (not .MacOSPackages) (.LinuxPackages) }}
+  {{- template "linux_packages" . }}
+  {{ end }}
```

Comment:
I think I didn't actually fix this situation properly yet, the Linux dependency is not implemented yet. And macOS needs a version defined. I think el_capitan is old enough to match everyone.
```suggestion
  {{ if and (.MacOSPackages) (not .LinuxPackages) }}
  depends_on macos: ">= :el_capitan"
  {{- template "macos_packages" . }}
  {{ end }}

  {{ if and (not .MacOSPackages) (.LinuxPackages) }}
  # depends_on :linux not implemented yet
  {{- template "linux_packages" . }}
  {{ end }}
```

Another option would be to just always do `on_macos` and `on_linux` blocks.

Reply:
yeah, we can probably remove these